File: src/lib-node/dir-read.js
function dir_read(x,y)
{
 check(is_str,x)
 
 if(is_undef(y))
  return dir_read(x,false)
 
 const r=arr()
 const dir=path_real(x)
 
 for(const v of fs.readdirSync(dir))
 {
  const s=path_concat(dir,v)
  
  if(is_file(s))
  {
   push(r,s)
   
   continue
  }
   
  if(y)
  {
   if(is_dir(s))
    push(r,s)
  }
 }
 
 return r
}

File: src/lib-node/dir/dir-read.cs
fn dir_read x:str with_dirs
 if is_undef with_dirs
  ret dir_read x false

 check is_bool with_dirs

 let r arr
 let dir path_real x
 let a fs.readdirSync dir

 sort a

 for a
  let s path_concat dir v

  if is_file s
   push r s

   cont
  end

  if with_dirs
   if is_dir s
    push r s
  end
 end

 ret r
end
